### PR TITLE
Don't throw exceptions for invalid baggage like other APIs

### PR DESCRIPTION
**Pull request by anuraaga**
I noticed that our baggage APIs validate parameters with exceptions, which isn't what we do in other APIs. This makes `Entry` creation package private (#1960 is a bit related) and ignores bad input in baggage construction.

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
Thanks!

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java [hunk 1]
```diff
@@ -135,6 +136,9 @@ public String toString() {
 
     @Override
     public Baggage.Builder put(String key, String value, EntryMetadata entryMetadata) {
+      if (!isKeyValid(key) || !isValueValid(value)) {
```
Comment (carlosalberto) [inline api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java:139]: Maybe we should do a fine-level logging here (can be discussed in/as a follow-up).

**Review by carlosalberto — APPROVED** [review]

File: api/src/main/java/io/opentelemetry/api/baggage/EntryMetadata.java [hunk 2]
```diff
@@ -26,6 +26,9 @@
    * @return an {@code EntryMetadata}.
    */
   public static EntryMetadata create(String metadata) {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/api/baggage/EntryMetadata.java:28]: Mark metadata as `Nullable`

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java:139]: why not throwing exception?

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java [hunk 3]
```diff
@@ -146,14 +150,14 @@ public String toString() {
...
-      return this;
+      return put(key, value, EntryMetadata.EMPTY);
     }
 
     @Override
     public Baggage.Builder remove(String key) {
+      if (key == null) {
+        return this;
+      }
       entries.remove(Objects.requireNonNull(key, "key"));
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java:161]: This is wrong, either not allow and check for null or throw exception.

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java [hunk 4]
```diff
@@ -135,6 +136,9 @@ public String toString() {
 
     @Override
     public Baggage.Builder put(String key, String value, EntryMetadata entryMetadata) {
+      if (!isKeyValid(key) || !isValueValid(value)) {
+        return this;
+      }
       entries.put(
           Objects.requireNonNull(key, "key"),
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java:143]: Either throw exception or allow `Nullable`

**Review by bogdandrutu — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/opentelemetry/api/baggage/EntryMetadata.java (hunk 2 above)
Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/api/baggage/EntryMetadata.java:28]: We don't use Nullable on APIs which shouldn't be passed null but we have defensive behavior anyways. IIRC this is the answer to all your comments?

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java (hunk 1 above)
Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java:139]: We currently don't log for parameter checks in most places I believe. Maybe we should, but I think we should do it consistently separate from removing the exceptions.

@bogdandrutu To throw exceptions, I believe we'd need a spec change https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles

File: api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java (hunk 3 above)
Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java]: Oops, removed the redundant checks

**Comment by jkwatson** [comment]
@anuraaga this needs a rebase. thanks!

**Comment by anuraaga** [comment]
@jkwatson Sorry for delay, finally rebased

**Pull request merged**